src/shims/x86/bmi.rs

```diff
@@ -0,0 +1,126 @@
+use rustc_span::Symbol;
+use rustc_target::spec::abi::Abi;
+
+use crate::*;
+
+impl<'tcx> EvalContextExt<'tcx> for crate::MiriInterpCx<'tcx> {}
+pub(super) trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
+    fn emulate_x86_bmi_intrinsic(
+        &mut self,
+        link_name: Symbol,
+        abi: Abi,
+        args: &[OpTy<'tcx>],
+        dest: &MPlaceTy<'tcx>,
+    ) -> InterpResult<'tcx, EmulateItemResult> {
+        let this = self.eval_context_mut();
+        // Prefix should have already been checked.
+        let unprefixed_name = link_name.as_str().strip_prefix("llvm.x86.bmi.").unwrap();
+
+        match unprefixed_name {
+            "pdep.32" => {
+                // <https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_pdep_u32>
+                this.expect_target_feature_for_intrinsic(link_name, "bmi2")?;
```

Comment:
llvm unhelpfully does not make a distinction between `bmi1` and `bmi2`: all bmi intrinsics are in the `bmi` namespace. So I don't see another way than to in every branch check for the relevant target features, even though it is error-prone.